REST: Unify query_string parameters parsing

javanna opened the pull request:
There currently are small differences between search api and count, exists, validate query, explain api when it comes to reading query_string parameters.  `analyze_wildcard`, `lowercase_expanded_terms` and `lenient` are only read by the search api and ignored by all other mentioned apis. Unified code to fix this and make sure it doesn't happen again. Also shared some code when it comes to printing out the query as part of SearchSourceBuilder conversion to ToXContent.

spinscale commented:
LGTM in general, but we may need to adapt the rest spec and add tests for the new APIs supporting the parameters?

javanna commented:
Thanks @spinscale I pushed a new commit. Updated REST spec (including parameters that were already support but not listed) and added REST tests. Tests were completely missing for count api and search_exists, so I added some basic generic tests too.

spinscale commented:
nice, ++ for more tests

LGTM
